students/guillaume/quarter_2_project/leankitbackend/leankitbackend/leankitcards.py

```diff
@@ -0,0 +1,297 @@
+            return ('').join([card.__repr__() for card in self.cards_list])
+        return 'There is no cards being imported'
+
+
+class Leankit_Card:
+    def __init__(self, card):
+        if isinstance(card, dict):
+            self.json = card
+            for key, value in card.items():
+                setattr(self, key, value)
+            self.status = True
+        else:
+            self.status = False
+
+    def __repr__(self):
+        try:
+            Name = ''
+            if len(self.assignedUsers) > 0:
+                tmp_lst = [user['fullName'] for user in self.assignedUsers]
+                Name = (', ').join(tmp_lst)
+            tmp_lst = self.type['title'], self.id, self.title, Name
+            return '{} {} {} {}\n'.format(*tmp_lst)
+        except (AttributeError, KeyError):
+            return 'input was not a card'
+
+
+class Leankit_Board:
+    def __init__(self, board):
+        if isinstance(board, dict):
+            self.json = board
+            for key, value in (self.json).items():
+                setattr(self, key, value)
+            self.status = True
+        else:
+            self.status = False
+
+    def __repr__(self):
+        try:
+            tmp_lst = [self.id, self.title]
+            ret_str = ' '.join(tmp_lst) + '\n'
+            ret_str += (('{}\n') * len(self.json)).format(*self.json.keys())
+            return ret_str
+        except AttributeError:
+            return 'Bad Data'
+
+    def users_list(self):
+        try:
+            return [user['fullName'] for user in self.users]
+        except (KeyError, AttributeError):
+            return []
+
+    def lanes_list(self):
+        def f(lane):
+            a = [lane['id'], lane['name'], lane['parentLaneId']]
+            return '{} {} {}'.format(*a)
+
+        try:
+            return [f(lane) for lane in self.lanes]
+        except (KeyError, AttributeError):
+            return []
+
+    def lanes_structures(self):
+
+        def f(key, lst):
+            if key in lst:
+                return lst.index(key)
+
+        def la(lane):
+            return [lane['name'], lane['id'], lane['index']]
+        try:
+            lst_1 = [la(lane) for lane in self.lanes
+                     if lane['parentLaneId'] is None]
+            d = defaultdict(list)
+            for lane in self.lanes:
+                if lane['parentLaneId'] is not None:
+                    d[lane['parentLaneId']].append(la(lane))
+
+            def g(key):
+                a = f(key, [item[1] for item in lst_1])
+                return lst_1[a][0]
+
+            e = {g(key): value for key, value in dict(d).items()}
+
+            for value in e.values():
+                value.sort(key=lambda x: x[-1])
+
+            return [lst_1, e]
+
+        except (KeyError, AttributeError):
+            return []
+
+
+def cards_per_types(cards, attribute='type', key='title'):
+    dic = defaultdict(list)
+    if isinstance(cards, Leankit_Cards):
```

Comment:
if you made Leankit_Cards an iterable, then this could jsut be:

`cards_lst - list(cards)`

with no type checking or anything.